miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultLibraryDilutionService.java

```diff
@@ -57,7 +60,34 @@ public LibraryDilution get(long dilutionId) throws IOException {
     return dilution;
   }
 
+  private boolean isTargetedSequencingRequired(Library library) {
+    return LimsUtils.isDetailedLibrary(library)
+        && ((DetailedLibrary) library).getLibraryDesignCode().isTargetedSequencingRequired();
+  }
+
+  private boolean isTargetedSequencingCompatible(TargetedSequencing ts, Library library) {
+    return library.getKitDescriptor().getTargetedSequencing().contains(ts);
+  }
+
+  private void validateTargetedSequencing(LibraryDilution dilution) {
+    TargetedSequencing ts = dilution.getTargetedSequencing();
+    Library library = dilution.getLibrary();
+
+    if (ts == null && isTargetedSequencingRequired(library)) {
+      throw new IllegalArgumentException("Targeted sequencing value is required");
+    } else if (LimsUtils.isDetailedLibrary(library) && !isTargetedSequencingCompatible(ts, library)) {
+      String tsName = ts.getAlias();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "ts" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)